spd/scripts/compare_models.py

```diff
@@ -0,0 +1,393 @@
+    """Compare two SPD models for geometric similarity between subcomponents."""
+
+    def __init__(
+        self,
+        current_model_path: str,
+        reference_model_path: str,
+        density_threshold: float = 0.0,
+        device: str = "auto",
+        comparison_config: dict[str, Any] | None = None,
+    ):
+        """Initialize the model comparator.
+
+        Args:
+            current_model_path: Path to current model (wandb: or local path)
+            reference_model_path: Path to reference model (wandb: or local path)
+            density_threshold: Minimum activation density for components to be included
+            device: Device to run comparison on ("auto", "cuda", "cpu")
+            comparison_config: Full comparison configuration dict
+        """
+        self.current_model_path = current_model_path
+        self.reference_model_path = reference_model_path
+        self.density_threshold = density_threshold
+        self.comparison_config = comparison_config or {}
+
+        if device == "auto":
+            self.device = get_device()
+        else:
+            self.device = device
+
+        logger.info(f"Loading current model from: {current_model_path}")
+        self.current_model, self.current_config = self._load_model_and_config(current_model_path)
+
+        logger.info(f"Loading reference model from: {reference_model_path}")
+        self.reference_model, self.reference_config = self._load_model_and_config(
+            reference_model_path
+        )
+
+    def _load_model_and_config(self, model_path: str) -> tuple[ComponentModel, dict[str, Any]]:
+        """Load model and config using the standard pattern from existing codebase."""
+        run_info = SPDRunInfo.from_path(model_path)
+        model = ComponentModel.from_run_info(run_info)
+        model.to(self.device)
+        model.eval()
+        model.requires_grad_(False)
+
+        config_dict = run_info.config.model_dump()
+
+        return model, config_dict
+
+    def create_eval_data_loader(self, config: dict[str, Any]) -> Iterator[Any]:
+        """Create evaluation data loader using exact same patterns as decomposition scripts."""
+        task_config = config.get("task_config", {})
+        task_name = task_config.get("task_name")
+
+        if not task_name:
+            raise ValueError("task_config.task_name must be set")
+
+        if task_name == "tms":
+            from spd.experiments.tms.models import TMSTargetRunInfo
+            from spd.utils.data_utils import DatasetGeneratedDataLoader, SparseFeatureDataset
+
+            if "pretrained_model_path" not in config or not config["pretrained_model_path"]:
+                raise ValueError("pretrained_model_path must be set for TMS models")
+
+            target_run_info = TMSTargetRunInfo.from_path(config["pretrained_model_path"])
+            n_features = target_run_info.config.tms_model_config.n_features
+            synced_inputs = target_run_info.config.synced_inputs
```

Comment:
I wouldn't bother creating these new variables if they're just used once and you can call them directly without adding new lines.

Reply:
I noticed these elsewhere in this PR, worth doing a read over.